s3-source-connector/src/main/java/io/aiven/kafka/connect/s3/source/utils/AWSV2SourceClient.java

```diff
@@ -128,19 +140,13 @@ public void addFailedObjectKeys(final String objectKey) {
         this.failedObjectKeys.add(objectKey);
     }
 
-    public void setFilterPredicate(final Predicate<S3Object> predicate) {
-        filterPredicate = predicate;
-    }
-
-    private boolean assignObjectToTask(final String objectKey) {
-        final int maxTasks = Integer.parseInt(s3SourceConfig.originals().get("tasks.max").toString());
-        final int taskId = Integer.parseInt(s3SourceConfig.originals().get("task.id").toString()) % maxTasks;
-        final int taskAssignment = Math.floorMod(objectKey.hashCode(), maxTasks);
-        return taskAssignment == taskId;
+    public void setFilterPredicate(final Predicate<S3Object> basePredicate) {
+        this.filterPredicate = basePredicate
```

Comment:
I think we do need this update to look something like the below and this way we just and additional predicates as we need them.
```suggestion
public void setFilterPredicate(final Predicate<S3Object> predicate) {
        this.filterPredicate = this.filterPredicate.and(predicate);
```

Reply:
This doesn't account for or predicates but we dont have a use case for that yet so I think it should be ok for the moment.